packages/neptune-extension/src/common/components/input/Input.tsx

```diff
@@ -0,0 +1,64 @@
+// Lib
+import React from 'react';
+import { bemBlock } from 'common/utils/bem';
+
+// Module
+import './Input.less';
+
+interface InputProps {
+  className?: string
+  disabled?: boolean
+  error?: boolean
+  selectOnFocus?: boolean
+  variant?: 'small' | 'big'
+  onFocus?: (ev?: React.FocusEvent<HTMLInputElement>) => void
+}
+
+const block = bemBlock('neptune-input');
+
+const Input: React.FC<InputProps & React.InputHTMLAttributes<HTMLInputElement>> = ({
+  className,
+  disabled,
+  error,
+  selectOnFocus,
+  variant,
+  onFocus,
+  ...props
+}) => {
+  const classNames = block({
+      modifiers: {
+        disabled,
+        error,
+        big: variant === 'big',
+      },
+      extra: className,
+    });
+
+  return (
+    <input
+      {...props}
+      className={classNames}
+      disabled={disabled}
+      onFocus={getOnFocusHandler(onFocus, selectOnFocus)}
+    />
+  );
+};
+
+function selectInputContent(ev: React.FocusEvent<HTMLInputElement>) {
+  ev.target.select();
+}
+
+function getOnFocusHandler(onFocus?: (ev: React.FocusEvent<HTMLInputElement>) => void, selectOnFocus?: boolean) {
+  if (selectOnFocus) {
```

Comment:
```suggestion
```suggestion
if (!selectOnFocus) {
  return onFocus;
}
if (!onFocus) {
  return event => event.target.select;
}
return () => {} // the combined case.
```
```

Reply:
I don't think your proposal is readable because you have to analyse each previous if statement to figure out if particular case was handled.

I propose something very explicit:
```ts
function getOnFocusHandler(onFocus?: FocusHandler, selectOnFocus?: boolean): FocusHandler {
  const handler: FocusHandler = (evt) => {
    if (selectOnFocus) {
      evt.target.select();
    }

    if (onFocus) {
      onFocus(evt);
    }
  };

  return handler;
}
```

Reply:
Ok, I'll change that